src/VisualStudio/Core/Def/Progression/GraphProvider.cs

```diff
@@ -189,14 +192,14 @@ public IEnumerable<GraphCommand> GetCommands(IEnumerable<GraphNode> nodes)
         }
 
         // All graph commands below this point apply only to Roslyn-owned nodes.
-        if (!nodes.All(n => IsRoslynNode(n)))
+        if (nodesInfo.ContainsNonRoslynNode)
         {
             yield break;
         }
 
         // Only show 'Base Types' and 'Derived Types' on a class or interface.
-        if (nodes.Any(n => IsAnySymbolKind(n, SymbolKind.NamedType) &&
-                           IsAnyTypeKind(n, TypeKind.Class, TypeKind.Interface, TypeKind.Struct, TypeKind.Enum, TypeKind.Delegate)))
+        if (nodesInfo.ContainsSymbolKindNamedType &&
+            (nodesInfo.ContainsTypeKindClassOrStruct || nodesInfo.ContainsTypeKindInterface || nodesInfo.ContainsTypeKindEnumOrDelegate))
```

Comment:
📝 The associativity of `&&` here seems to have changed

Reply:
I'm not seeing the functional differences in the spots you've mentioned. Maybe an explicit example of an input that would have changed behavior would be the best way for me to understand your concern.

Reply:
Oh, I see now. All callers I've encountered have called this with a single item in the array, but if there were multiple items, then these conditions are no longer such that a single item would have needed met the criteria.